.github/workflows/build-deploy-docs.yml

```diff
@@ -40,13 +45,16 @@ jobs:
       - name: Build docs
         run: |
           just build-docs
-      - name: Upload artifact
+      - name: Upload pages artifact
         uses: actions/upload-pages-artifact@v3
         with:
           path: ./site
 
   # Deployment job
   deploy:
+    name: Deploy github pages
+    # Disallow running deployment for pull requests
+    if: github.event_name != 'pull_request'
```

Comment:
Could consider something like:

```
if: github.repository_owner == 'nominal-io' && github.ref == 'refs/heads/main'
```

To avoid running deploys from forked repo branches on push to main.